slick/src/main/scala/slick/ast/ClientSideOp.scala

```diff
@@ -35,6 +35,28 @@ final case class First(val child: Node) extends UnaryNode with SimplyTypedNode w
   def nodeMapServerSide(keepType: Boolean, r: Node => Node) = mapChildren(r, keepType)
 }
 
+final case class FirstOption(val child: Node) extends UnaryNode with SimplyTypedNode with ClientSideOp {
+  type Self = FirstOption
+  protected[this] def rebuild(ch: Node) = copy(child = ch)
+  lazy val innerType = child.nodeType.asCollectionType.elementType
+  protected def buildType = OptionType(innerType)
+  def nodeMapServerSide(keepType: Boolean, r: Node => Node) = {
+    val this2 = mapChildren(r, keepType)
+    // This is to get around the UnassignedType of the child after inferTypes
+    // TableExpansion nodes have UnassignedType after InferTypes phase
+    // and ExpandTables.mapServerSide preserves type.
+    // This leads to a type mismatch after ExpandTables
+    if (this.child.nodeType eq this2.child.nodeType) this2 else this2.untyped.infer()
```

Comment:
This looks suspicious. I don't think there should be an `UnassignedType` anywhere after `inferTypes`, so if you encounter one it's probably a bug somewhere else and there's a proper way to fix it instead of retyping it here.